### DISCUSSION TITLE
is there if else block to run commands?

### DISCUSSION THREAD

**Discussion by FischLu** [zenangst/KeyboardCowboy#575] [General]
Hi,

I found this project really cool, and I think it has the potential to replace my Hammerspoon scripts. However, I didn't notice any support for if-else blocks when running commands. For example, I want to set a shortcut key, but only execute the command when certain conditions are met. Another use case is having the shortcut key work only when the currently focused application is the one I specify.

If these features are not currently available, I'm willing to try implementing them myself, but I'd like to know if you'd be interested in accepting such functionality.

**Comment by zenangst** [comment]
Hi @FischLu,

Unfortunately, there's no pure 'if' statement functionality available right now. I've thought about how it could work in the past, but I haven't found a good or practical way to implement it yet.

However, you can achieve scoping workflows to only be available when a specific application is the frontmost one by using Groups that are scoped to that application (one or many).

You can learn more about groups in the wiki: https://github.com/zenangst/KeyboardCowboy/wiki/Groups#the-role-of-groups

**Reply by FischLu** [reply]
Thank you for the information of Groups, it works fine for scoping workflow!
